rust/src/transformations/polars/make_col/mod.rs

```diff
@@ -0,0 +1,191 @@
+use num::One;
+use opendp_derive::bootstrap;
+use polars::prelude::*;
+use std::collections::BTreeSet;
+
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{Context, ExprDomain, ExprMetric, LazyFrameDomain};
+use crate::error::*;
+use crate::traits::DistanceConstant;
+
+#[bootstrap(ffi = false)]
+/// Make a Transformation that returns a col(<column_name>) expression for a Lazy Frame.
+///
+/// # Arguments
+/// * `input_domain` - Domain of the expression to be applied.
+/// * `input_metric` - The metric space under which neighboring LazyFrames are compared.
+///
+/// # Generics
+/// * `M` - Dataset Metric type.
+/// * `C` - Context in which expression is applied.
+///
+pub fn make_col<M, C: Context>(
+    input_domain: ExprDomain<C>,
+    input_metric: M,
+    col_name: String,
+) -> Fallible<Transformation<ExprDomain<C>, ExprDomain<C>, M, M>>
+where
+    M: ExprMetric<C>,
+    M::Distance: DistanceConstant<M::Distance> + One + Clone,
+    (ExprDomain<C>, M): MetricSpace,
+{
+    if input_domain
+        .lazy_frame_domain
+        .column(col_name.as_str())
+        .is_none()
+    {
+        return fallible!(MakeTransformation, "{} is not in dataframe", col_name);
+    }
+
+    let context_columns = input_domain.context.grouping_columns();
+
+    let mut columns_to_keep = BTreeSet::from_iter(context_columns);
+    columns_to_keep.insert(col_name.clone());
+
+    let series_domains = input_domain
+        .clone()
+        .lazy_frame_domain
+        .series_domains
+        .into_iter()
+        .filter(|s| columns_to_keep.contains(&s.field.name.to_string()))
+        .collect();
+
+    let margins = input_domain
+        .clone()
+        .lazy_frame_domain
+        .margins
+        .into_iter()
+        .filter(|(s, _)| s.is_subset(&columns_to_keep))
+        .collect();
+
+    let output_domain = ExprDomain {
+        lazy_frame_domain: LazyFrameDomain {
+            series_domains,
+            margins,
+        },
+        context: input_domain.context.clone(),
+        active_column: Some(col_name.clone()),
+    };
+
+    Transformation::new(
+        input_domain.clone(),
+        output_domain,
+        Function::new_fallible(
+            move |(frame, expr): &(C::Value, Expr)| -> Fallible<(C::Value, Expr)> {
+                if expr != &Expr::Wildcard {
+                    return fallible!(
+                        FailedFunction,
+                        "make_col has to be the first expression in the expression chain"
+                    );
+                }
+                Ok((frame.clone(), col(&*col_name)))
+            },
+        ),
+        input_metric.clone(),
+        input_metric,
+        StabilityMap::new_from_constant(M::Distance::one()),
```

Comment:
Minor nit, but can you change this to `StabilityMap::new(Clone::clone)`? Then the trait bound above can just be `M::Distance: Clone,`